### ISSUE TITLE
Unsupported options on Arch instructions

### ISSUE THREAD

**Issue by NEXCydran** [cockpit-project/cockpit-project.github.io#313]
Hello,
The instructions for [installing Cockpit on Arch](https://cockpit-project.org/running.html#archlinux) currently uses unsupported options for `pacman`:
```
sudo pacman -Sy cockpit
sudo systemctl enable --now cockpit.socket
```

Using `pacman -Sy` causes a partial upgrade, which is [unsupported](https://wiki.archlinux.org/index.php/System_maintenance#Partial_upgrades_are_unsupported) and could potentially cause breakage (the new package will pull any updated libraries, which may break packages that haven't been updated yet).

I recommend updating the instructions to:
```
sudo pacman -Syu && sudo pacman -S cockpit
sudo systemctl enable --now cockpit.socket
```

This will perform a full upgrade (`pacman -Syu`) first, then install Cockpit (`pacman -S cockpit`).

**Comment by martinpitt** [comment]
Hello @NEXCydran , thanks for reviewing this! Indeed it wasn't my intention to suggest any upgrade at all, I just wanted this to be a simple install. That's more or less what I found with googling, but I have zero knowledge about pacman. Is there a more standard command that just installs a new package, without side effects?

**Comment by NEXCydran** [comment]
Hello @martinpitt,
`pacman -S cockpit` 
is the standard command for installing a package without any other side effects.

**Comment by martinpitt** [comment]
I still don't understand how `-y` would install any upgrades. `pacman -S -h` says
```
  -y, --refresh        download fresh package databases from the server
                       (-yy to force a refresh even if up to date)
```

I actually did start with `pacman -S cockpit`, but on a fresh system that just fails with
```
# pacman -S cockpit
warning: database file for 'core' does not exist (use '-Sy' to download)
warning: database file for 'extra' does not exist (use '-Sy' to download)
warning: database file for 'community' does not exist (use '-Sy' to download)
error: target not found: cockpit
```
So it advertises `-Sy` by itself, and the operation doesn't seem to upgrade anything else, at least on my small test VM. Whereas `pacman -Syu` actually does upgrade packages, but that's not really what I want to document.

Honestly this is all rather confusing to me, coming from a Debian, Ubuntu, and now Fedora background.. There the operations of "fetch up to date package indexes", "update system", and "install new package" are distinct. Of course installing a new package *might* upgrade one as a side effect if it has corresponding dependencies, but that's then exactly right?

So `pacman -S cockpit` is clearly not sufficient here..

**Comment by NEXCydran** [comment]
Indeed, `pacman -Sy` updates the database (like `apt update`), which is why `pacman` recommends it, but the developers expect this to be followed with an upgrade (`-Su`, like `apt upgrade`).

Unlike `apt` or `yum`, `pacman` is not made to handle updating the caches w/o updating the packages as well (partial upgrade), as Arch is a rolling-release distro (packages can be updated at any time, and must be rebuilt upon library upgrades). The 'flow' of `pacman` requires a full upgrade, then an install (`pacman -Syu && pacman -S cockpit`).

It is expected that the user would update _and_ upgrade their system regularly, as part of [system maintenance](https://wiki.archlinux.org/index.php/System_maintenance), before installing packages with `pacman -S`.

Installing a new package _might_ upgrade another one of course, but if it upgrades a library, any packages that rely on that library will be broken until it is rebuilt again by `pacman`. To avoid that problem, users are expected to upgrade everything first, then install the package.

As a sidenote, `pacman` allows users to 'combine' operations into one command for convience; this is usually done with updating and upgrading (`pacman -Syu = pacman -Sy && pacman -Su`). In addition, the Arch Wiki does have a [page that compares `pacman` to other package managers (`apt`, `dnf`, etc..)](https://wiki.archlinux.org/index.php/Pacman/Rosetta) by 'translating' `pacman` commands to their equivalents.

I've debated on including `pacman -Syu` into the directions or not - updating & upgrading is not a specific requirement to install Cockpit, but it is expected that a user would run `pacman -Syu` regularly, and not doing so can cause problems (like the error above).

**Comment by martinpitt** [comment]
Thanks @NEXCydran for the details! So I suppose documenting just `-S` is fine, and adding a note to run `-Syu` if the cache is not present should cover the cases?

**Comment by NEXCydran** [comment]
Yep, I think that would suffice.

**Comment by martinpitt** [comment]
Fixed in PR #314
